plane2/src/drone/backend_manager.rs

```diff
@@ -38,17 +41,36 @@ impl StepStatusResult {
     }
 }
 
-type StateCallback = Box<dyn Fn(&BackendState) -> Result<(), Box<dyn Error>> + Send + Sync>;
+struct Callback<T>(Box<dyn Fn(T) -> Result<(), Box<dyn Error>> + Send + Sync>);
```

Comment:
Is there an advantage to using a newtype here over a type alias? Is it that you wanted Debug implemented?

Reply:
So i can stick tracing::instrument on things to debug.